### PR TITLE
fix: apply correlation between groups

### PR DESCRIPTION
**Pull request by e-fisher**
## Description

Fixes a bug where correlation rules wouldn't be applied correctly between groups. Also, I've added grouped recording to the unit test to reproduce the regression. 

## How to Test
[Archive.zip](https://github.com/user-attachments/files/17517390/Archive.zip)

Download attached recording and generator, verify:
- Correlation value is extracted only once
- Correlated value is applied to post body
- Group are added correctly

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`npm run lint`) and all checks pass.
- [x] I have run tests locally (`npm test`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: src/codegen/codegen.utils.ts
Comment (e-fisher) [inline src/codegen/codegen.utils.ts:120]: This was implicitly typed as `any`

File: src/codegen/codegen.ts
Comment (e-fisher) [inline src/codegen/codegen.ts:16]: Accept ungrouped proxy data to apply rules to every request and group it after

File: src/codegen/codegen.ts
Comment (Llandy3d) [inline src/codegen/codegen.ts:64]: typo! 😄

**Review by Llandy3d — DISMISSED** [review]
Reviewing from the functionality standpoint, it seems to have fixed the issue 🚀 

I would wait for another review that can review better the code 🙇

**Review by going-confetti — APPROVED** [review]
LGTM 🚀

**Pull request merged**
